migrations/20240729123726_add_mfa_phone_config.up.sql

```diff
@@ -11,4 +11,4 @@ alter table {{ index .Options "Namespace" }}.mfa_challenges add column if not ex
 
 create index if not exists idx_sent_at on {{ index .Options "Namespace" }}.mfa_challenges(sent_at);
 
-create unique index unique_verified_phone_factor on {{ index .Options "Namespace" }}.mfa_factors (user_id, phone) where status = 'verified';
+create unique index if not exists unique_verified_phone_factor on {{ index .Options "Namespace" }}.mfa_factors (user_id, phone) where status = 'verified';
```

Comment:
The consideration here was to allow for multiple calls to enroll with phone so the dev can verify any one of the unverified factors instead of searching for the sole unverified factor

Open to dropping the constraint but I would want to return the existing phone factor if a phone enroll is called multiple times so the developer doesn't have to search for it

Alternatively, can return an error and ask the developer to call `listFactors` or similar

Reply:
> The consideration here was to allow for multiple calls to enroll with phone so the dev can verify any one of the unverified factors instead of searching for the sole unverified factor

Do we clean up the rest of the unverified factors upon verification?

> Open to dropping the constraint but would want to return the existing phone factor if a phone enroll is called multiple times so the developer doesn't have to search for it

Hmm the current behaviour where we return a new factor each time is consistent with the TOTP behaviour so it's nice to keep it. However, i prefer having a unique constraint over a partial unique index 🤔 

cc @hf wdyt?

Reply:
> Do we clean up the rest of the unverified factors upon verification?
Yup, unverified factors [should be deleted on successful verification](https://github.com/supabase/auth/blob/master/internal/api/mfa.go#L491)

Reply:
Hmm... yeah I've also missed this. Let's not do partial unique indexes unless absolutely necessary. They bring very little benefit and just bloat the database.

Let's just do unique on `(user_id, phone)` and that should be it.

Reply:
So we return an error if a user attempts to return a phone number twice?

Reply:
if the phone number isn't verified yet, then calling enroll multiple times with the same phone number should always replace the old factor with a new one

if the phone number has been verified, then calling enroll again with the same phone number should return an error

we can then add a unique constraint on (user_id, phone)

Reply:
Sounds good, let's do that

Reply:
opening a FLUP with the changes to regenerate / check for existing phone factor so that we can isolate the changes from this one.